superkey/forged_application.go

```diff
@@ -135,3 +137,22 @@ func (f *ForgedApplication) applicationExtraPayload() map[string]interface{} {
 
 	return extra
 }
+
+const DEFAULT_SLEEP_TIME = 7
+
+// read from the ENV first - if there isn't anything there fall back to the old
+// default which is 7 seconds. defined ^^
+func waitTime() time.Duration {
+	raw := os.Getenv("AWS_WAIT_TIME")
+	if raw == "" {
+		return DEFAULT_SLEEP_TIME // chosen by fair dice roll
+	}
+
+	i, err := strconv.ParseInt(raw, 10, 64)
+	if err != nil {
+		l.Log.Errorf("Failed to parse %q as sleep time - defaulting to %v", raw, DEFAULT_SLEEP_TIME)
+		return DEFAULT_SLEEP_TIME
```

Comment:
Here the `DEFAULT_SLEEP_TIME` is returned directly. Shouldn't it be wrapped as `time.Duration(DEFAULT_SLEEP_TIME)`? I guess not, because the checks passed, but I'm really curious

Reply:
So this is where it gets interesting - since it's a constant (known at compile time) the Go compiler is smart enough to know that we can use it as `time.Duration`, since `time.Duration` is just an `int64` [under the hood](https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/time/time.go;l=590).

Basically you have to explicitly cast variables - but don't have to cast them if they're constants. That is why you can write:
```go
time.Sleep(7 * time.Second)
```
but not
```go
x := 2
time.Sleep(x * time.Second) // won't work
time.Sleep(time.Duration(x) * time.Second) // will work
```

Hopefully clear as mud. Basically the compiler is auto-converting the constant to the right type.

Reply:
Absolutely clear, thanks for the explanation!